airflow/ui/src/components/TriggerDag/TriggerDAGForm.tsx

```diff
@@ -0,0 +1,184 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Input, Button, Box, Text, Spacer, HStack } from "@chakra-ui/react";
+import { json } from "@codemirror/lang-json";
+import { githubLight, githubDark } from "@uiw/codemirror-themes-all";
+import CodeMirror from "@uiw/react-codemirror";
+import { useEffect, useState } from "react";
+import { useForm, Controller } from "react-hook-form";
+
+import { useColorMode } from "src/context/colorMode";
+
+import { Accordion } from "../ui";
+import type { DagParams } from "./TriggerDag";
+
+type TriggerDAGFormProps = {
+  dagParams: DagParams;
+  onClose: () => void;
+  onTrigger: (updatedDagParams: DagParams) => void;
+  setDagParams: React.Dispatch<React.SetStateAction<DagParams>>;
+};
+
+const TriggerDAGForm: React.FC<TriggerDAGFormProps> = ({
+  dagParams,
+  onTrigger,
+  setDagParams,
+}) => {
+  const [jsonError, setJsonError] = useState<string | undefined>();
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+  } = useForm<DagParams>({
+    defaultValues: dagParams,
+  });
+
+  useEffect(() => {
+    reset(dagParams);
+  }, [dagParams, reset]);
+
+  const onSubmit = (data: DagParams) => {
+    onTrigger(data);
+    setDagParams(data);
+    setJsonError(undefined);
+  };
+
+  const validateAndPrettifyJson = (value: string) => {
+    try {
+      const parsedJson = JSON.parse(value) as JSON;
+
+      setJsonError(undefined);
+
+      return JSON.stringify(parsedJson, undefined, 2);
+    } catch {
+      setJsonError("Invalid JSON format.");
```

Comment:
Small improvement - which makes it easier for users: Can you add the parsing error itself which is in the exception text? Then a user knows better where to fix.

Reply:
Sure, that will be user friendly.

Reply:
Updated UI:
<img width="927" alt="image" src="https://github.com/user-attachments/assets/6dc537de-eb97-43fe-81a9-1a3b249eee41">